test/gmp.jl

```diff
@@ -542,3 +542,161 @@ end
+    @test a+BigInt(1) == b
+    @test typeof(a+1) == Rational{BigInt}
+    @test a+1 == b
+    @test isequal(a+1, b)
+    @test b == a+1
+    @test !(b == a)
+    @test b > a
+    @test b >= a
+    @test !(b < a)
+    @test !(b <= a)
+
+    @test typeof(a * 2) == Rational{BigInt}
+    @test a*2 == c
+    @test c-a == a
+    @test c == a + a
+    @test c+1 == a+b
+
+    @test typeof(d) == Rational{BigInt}
+    @test d == -c
+
+
+    @test e == a // b
+
+    @testset "gmp cmp" begin
+        @test Base.GMP.MPQ.cmp(b,   a) ==  1
+        @test Base.GMP.MPQ.cmp(a,   b) == -1
+        @test Base.GMP.MPQ.cmp(c-a, a) ==  0
+    end
+
+    @testset "division errors" begin
+        oz = Rational{BigInt}(0, 1)
+        zo = Rational{BigInt}(1, 0)
+
+        @test oz + oz == 2 * oz
+        @test oz // zo == oz
+        @test zo // oz == zo
+
+        @test_throws DivideError() zo - zo
+        @test_throws DivideError() zo + (-zo)
+        @test_throws DivideError() zo * oz
+        @test_throws DivideError() oz // oz
+        @test_throws DivideError() zo // zo
+    end
+
+    @testset "big infinities" begin
+        oz   = Rational{BigInt}(1, 0)
+        zo   = Rational{BigInt}(0, 1)
+        o    = Rational{BigInt}(1, 1)
+
+        @test oz + zo    == oz
+        @test zo - oz    == -oz
+        @test zo + (-oz) == -oz
+        @test -oz + zo   == -oz
+
+        @test (-oz) * (-oz) == oz
+        @test (-oz) * oz    == -oz
+
+        @test o // zo       == oz
+        @test (-o) // zo    == -oz
+    end
+end
+
+
+a = Rational{BigInt}(1, 2)
+b = Rational{BigInt}(-1, 3)
+c = Rational{BigInt}(3, 2)
+t = Rational{BigInt}(0, 1)
+aa = copy(a)
+bb = copy(b)
+cc = copy(c)
+@testset "big rational inplace" begin
+    @test Base.GMP.MPQ.add!(t, a, b) == Rational{BigInt}(1, 6)
+    @test t == Rational{BigInt}(1, 6)
+    @test Base.GMP.MPQ.add!(t, t) == Rational{BigInt}(1, 3)
+    @test t == Rational{BigInt}(1, 3)
+
+    @test iszero(Base.GMP.MPQ.sub!(t, t))
+    @test iszero(t)
+    @test Base.GMP.MPQ.sub!(t, b, c) == Rational{BigInt}(-11, 6)
+    @test t == Rational{BigInt}(-11, 6)
+
+    @test Base.GMP.MPQ.mul!(t, a, b) == Rational{BigInt}(-1, 6)
+    @test t == Rational{BigInt}(-1, 6)
+    @test Base.GMP.MPQ.mul!(t, t) == Rational{BigInt}(1, 36)
+    @test t == Rational{BigInt}(1, 36)
+    @test iszero(Base.GMP.MPQ.mul!(t, Rational{BigInt}(0)))
+
+    @test Base.GMP.MPQ.div!(t, a, b) == Rational{BigInt}(-3, 2)
+    @test t == Rational{BigInt}(-3, 2)
+    @test Base.GMP.MPQ.div!(t, a) == Rational{BigInt}(-3, 1)
+    @test t == Rational{BigInt}(-3, 1)
+
+    @test aa == a && bb == b && cc == c
+
+    @testset "set" begin
+        @test Base.GMP.MPQ.set!(a, b) == b
+        @test a == b == bb
+
+        Base.GMP.MPQ.add!(a, b, c)
+        @test b == bb
+
+        @test Base.GMP.MPQ.set_z!(a, BigInt(0)) == BigInt(0)
+        @test iszero(a)
+        @test Base.GMP.MPQ.set_z!(a, BigInt(3)) == BigInt(3)
+        @test a == BigInt(3)
```

Comment:
Same thing as above, you can replace `== BigInt(0)` by `== 0` and `== BigInt(3)` by `== 3`.